PayPalCheckout/src/main/java/com/paypal/android/checkout/PayPalConfiguration.kt

```diff
@@ -0,0 +1,73 @@
+package com.paypal.android.checkout
+
+import android.app.Application
+import com.paypal.android.core.Environment
+import com.paypal.android.core.CoreConfig
+
+class PayPalConfiguration(
+    /**
+     * Client ID is used to connect the Checkout SDK to your PayPal app. Your app's Client ID can be
+     * found in the PayPal Developer Portal.
+     *
+     * @see [Developer Portal](https://developer.paypal.com/developer/applications/)
+     * @see [Managing Sandbox Apps]
+     * (https://developer.paypal.com/docs/api-basics/manage-apps/#create-or-edit-sandbox-and-live-apps)
+     */
+    clientId: String,
+
+    /**
+     * Sets the [Environment] that the SDK will run in.
+     *
+     * Use [Environment.LIVE] for production.
+     * Use [Environment.SANDBOX] for development and testing.
+     */
+    environment: Environment,
```

Comment:
I think merchants should already have provided `clientId` and `environment` in `CoreConfig`, so having them here might be confusing.
What do you think of removing `PayPalConfiguration`, and move other necessary params for PayPal to `PayPalClient`?

Reply:
These fields are available on `CoreConfig` in iOS we only need `clientID`, `returnUrl`, and `environment` can we do the same here? We can still encapsulate these fields within `PayPalClient`.

Reply:
so are we going to dispose of any "modular" configuration? Like all of our clients will be configured with a `CoreConfig` and any other field will be passed as parameter? Something like this:
`var client = GenericClient(CoreConfig(clientId, enviornment), field1, field2,...)`

Reply:
We may need it eventually but we can leave it out for now. I believe the collective understanding is that we will incrementally add functionality to configure native checkout.

Reply:
One thing to note here is we technically don't need to call `PayPalCheckout.setConfig` until we call `PayPalClient.checkout`. Some of these additional configuration items could be moved to a `PayPalRequest` object and applied when a merchant starts the Native XO flow.

Reply:
@sshropshire well not quite. In Android, `PayPalCheckout.setConfig` performs a request, that needs to be completed before starting the nxo flow. Its a bad design but we are in talks to remove and change that. But for now, merchants needs to have the possibility to `setConfig` independently

Reply:
Got it. Are there defaults for these pieces of configuration? Since this is going to change soon we could go with `clientID`, `environment`, and `returnUrl` and then work on reducing the size of this PR.